src/volume.ts

```diff
@@ -221,6 +221,32 @@ export class Volume extends Construct implements IStorage {
     });
   }
 
+  /**
+   * Populate the volume from a SecretProviderClass created as part of a K8s
+   * Secrets Store CSI Driver installation:
+   * https://secrets-store-csi-driver.sigs.k8s.io/introduction.html.
+   *
+   * The Secret Store CSI driver will need an associated provider to source the
+   * secrets. The following provider allows sourcing from AWS Secrets Manager
+   * and Systems Manager Parameter Store:
+   * https://aws.github.io/secrets-store-csi-driver-provider-aws/
+   *
+   * @param className The name of the secret provider class to use to populate the volume.
+   * @param options Options
+   */
+  public static fromSecretProviderClass(scope: Construct, id: string, className: string, options: SecretProviderVolumeOptions = { }): Volume {
```

Comment:
So my concern with this is that we are creating a coupling with specifications we have no control over, that are decoupled from the kubernetes spec itself. 

Imagine that `secrets-store` change either the schema (i.e the `secretProviderClass` property name or structure), or the driver name (i.e `secrets-store.csi.k8s.io`) - what would this function do? it would have to be incompatible with a certain version of the driver. 

I agree that this approach is better that accepting a `SecretProviderClass` object, but the underlying concern still holds true.

This is why integrations with third-party components should be implemented as third-party libraries (ideally published by the vendor itself). The problem of course is that such integrations are currently impossible because the `Volume` class is locked for extensions, but we will address this issue separately. 

In the meantime, I suggest we change this function to support the generic `csi` volume instead. Basically:

```suggestion
  public static fromCsi(scope: Construct, id: string, driver: string, options: CsiVolumeOptions = { }): Volume {
```

Where `CsiVolumeOptions` will contain an `attributes` property to hold the driver specific options.

Reply:
Done!